src/MailChimp/Facade.php

```diff
@@ -350,6 +350,43 @@ public function handle_grouping_integrity( $local_groups, $remote_groups, $group
 	}
 
 	/**
+	 * Create all segments necessary
+	 *
+	 * @param array $segments an array of segments
+	 * @return bool
+	 */
+	public function create_segments_from_groups( $segments ) {
+
+		foreach ( $segments as $segment ) {
+
+			try {
+
+				$result = $this->facade->lists->segmentTest( $this->current_list['id'], $segment );
+
+				if ( $result ) {
+
+					// Try adding the segment
+					$args = [
+						'type'         => 'saved',
+						'name'         => md5( $segment['conditions'][0]['value'] ), // 100 byte max so md5 it is
```

Comment:
> I was planning on recreating all segment if we need an update as it would be a little easier to manage

You mean, on the slightest change (say a category is no longer made available), delete all Segments and recreate them? That would require reconfiguring all the Campaigns to use the new Segments. See my email (2015-02-27 @ 10:50) for ideas regarding it. I didn't see this comment until after sending the email so certain pieces are obsolete, notably:

> Les renommés afin d'être lisible—e.g., Business Payment,Cryptocurrency,...—deviendrait ridiculement long à une combinaison de 7 catégories et plus. [...]

This part, though, could be a way to manage this:

[...] Peut-être qu'on pourrait avoir une page, sur (ou comme) "Overview" qui servirait à lister toutes les combinaisons et leur hash sur MailChimp (et un lien)?